### PR TITLE
Add tranlations for texts and links. Format error messages in red.

### PR DESCRIPTION
**Pull request by liviuignat**
- Added translations for all pages
- Added `FormattedLink` component to be able to add also the link href into translations
- Formatted server-side error messages

### CODE REVIEW

File: src/translations/locale/locale.en-US.json [hunk 1]
```diff
@@ -1,4 +1,27 @@
 {
+  "App.about.us": "About Us",
+  "App.company": "Company",
+  "App.imprint": "Imprint",
+  "App.privacy.policy": "Privacy policy",
+  "App.signup": "Sign up",
+  "App.login": "Login",
+  "App.email": "E-Mail",
+  "App.password": "Passwort",
```
Comment (RaulTsc) [inline src/translations/locale/locale.en-US.json]: Password

File: src/translations/locale/locale.en-US.json [hunk 2]
```diff
@@ -1,4 +1,27 @@
...
+  "App.link.about.us": "https://www.everreal.co/en-US/about-us",
+  "App.link.imprint": "https://www.everreal.co/en-US/imprint",
+  "App.link.privacy.policy": "https://www.everreal.co/en-US/privacy-policy",
```
Comment (RaulTsc) [inline src/translations/locale/locale.en-US.json]: "App.link.privacy.policy": "{baseUrl}/en-US/privacy-policy" maybe?

File: src/translations/locale/locale.en-US.json [hunk 3]
```diff
@@ -1,4 +1,27 @@
...
+  "LoginPage.page.title": "Login to EverReal",
+  "RegisterPage.page.title": "Sign up to EverReal",
+  "RegisterPage.label.email": "Your email",
+  "RegisterPage.label.password": "Create a password",
+  "RegisterPage.label.repeat.password": "Confirm your password",
+  "ResetPassword.page.title": "Did you forget your password?",
```
Comment (RaulTsc) [inline src/translations/locale/locale.en-US.json]: ResetPasswordPage.page.title

File: src/universal/common/components/AppFooter/AppFooter.tsx [hunk 4]
```diff
@@ -1,56 +1,65 @@
...
-            </div>
-            <ul className={css.AppFooter_sectionLinks}>
-              <li><a href="#">About us</a></li>
-              <li><a href="#">Imprint</a></li>
-              <li><a href="#">Privacy policy</a></li>
-            </ul>
-          </div>
+          <ul className={css.AppFooter_sectionLinks}>
+            <li>
+              <a href="/de-DE/login" title="Deutsch">
```
Comment (RaulTsc) [inline src/universal/common/components/AppFooter/AppFooter.tsx]: Should not be hardcoded. It should be something like "/de-DE/{path}" as we are using this on all pages. i.e register & resetpassword pages both point to /login.

File: src/universal/common/components/AppFooter/AppFooter.tsx [hunk 5]
```diff
@@ -1,56 +1,65 @@
...
+                <img src="https://s3.eu-central-1.amazonaws.com/s3-production/assets/images/common/flags/de.png" alt="de" />&nbsp;Deutsch
+              </a>
+            </li>
+            <li>
+              <a href="/en-US/login" title="English">
```
Comment (RaulTsc) [inline src/universal/common/components/AppFooter/AppFooter.tsx]: Same as above

File: src/universal/common/components/AppHeader/AppHeader.tsx [hunk 6]
```diff
@@ -9,7 +9,7 @@ export default class AppHeader extends React.Component<any, any> {
     return (
       <div className={css.AppHeader}>
         <div className={css.LogoContainer}>
-          <a href="/"><AppLogo /></a>
+          <a href="https://www.everreal.de"><AppLogo /></a>
```
Comment (RaulTsc) [inline src/universal/common/components/AppHeader/AppHeader.tsx]: ?

File: src/translations/locale/locale.de-DE.json [hunk 7]
```diff
@@ -1,4 +1,27 @@
...
+  "App.company": "Unternehmen",
+  "App.imprint": "Impressum",
+  "App.privacy.policy": "Datenschutz",
+  "App.signup": "Registrieren",
+  "App.login": "Anmelden",
+  "App.email": "E-Mail",
+  "App.password": "Passwort",
+  "App.link.about.us": "https://www.everreal.co/de-DE/ueber-uns",
+  "App.link.imprint": "https://www.everreal.co/de-DE/impressum",
+  "App.link.privacy.policy": "https://www.everreal.co/de-DE/datenschutz",
```
Comment (RaulTsc) [inline src/translations/locale/locale.de-DE.json]: Same as en translation.

File: src/universal/common/components/FormattedLink/FormattedLink.tsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+interface IFormattedLinkProps {
+  id: string;
+  children: any;
+}
+
+export const FormattedLink = (props: IFormattedLinkProps, context) => {
+  const {intl: {formatMessage}} = context;
+  const {id, children} = props;
+  const formattedHref = formatMessage({id});
```
Comment (RaulTsc) [inline src/universal/common/components/FormattedLink/FormattedLink.tsx]: id & baseUrl

**Comment by RaulTsc** [comment]
Added some small comments. We should fix them before merging. Other than that, looks great! 👍

**Comment by liviuignat** [comment]
Awesome, thx!

File: src/translations/locale/locale.en-US.json (hunk 2 above)
Comment (liviuignat) [inline src/translations/locale/locale.en-US.json]: Where would you see that `baseUrl` should be coming from? Keep in mind that this will be an authentication/authorization micro-service that will be hosted on a subdomain (like auth.everreal.co), and these links should point out to the current website that will host these pages. Do you mean that `baseUrl` should be added in configuration?

File: src/translations/locale/locale.en-US.json (hunk 1 above)
Comment (liviuignat) [inline src/translations/locale/locale.en-US.json]: Yep, thx.

File: src/translations/locale/locale.en-US.json (hunk 3 above)
Comment (liviuignat) [inline src/translations/locale/locale.en-US.json]: Yep, thx!

File: src/universal/common/components/AppFooter/AppFooter.tsx (hunk 4 above)
Comment (liviuignat) [inline src/universal/common/components/AppFooter/AppFooter.tsx]: Indeed. Will try to see and think if I could read the partial path. If it takes too long I will add a technical task and save it for later.

File: src/universal/common/components/AppHeader/AppHeader.tsx (hunk 6 above)
Comment (liviuignat) [inline src/universal/common/components/AppHeader/AppHeader.tsx]: This has to point to the main website, `/` will point to `https://auth.everreal.co`, but should point to `https://www.everreal.co`.

File: src/universal/common/components/FormattedLink/FormattedLink.tsx (hunk 8 above)
Comment (liviuignat) [inline src/universal/common/components/FormattedLink/FormattedLink.tsx]: same as above, not sure where you would see `baseUrl` coming from.

File: src/universal/common/components/AppHeader/AppHeader.tsx (hunk 6 above)
Comment (RaulTsc) [inline src/universal/common/components/AppHeader/AppHeader.tsx]: Cool, so then it should point to `.co` not to `.de` right?

File: src/universal/common/components/AppFooter/AppFooter.tsx (hunk 4 above)
Comment (RaulTsc) [inline src/universal/common/components/AppFooter/AppFooter.tsx]: Maybe something like `const regExp = new RegExp("\/[a-zA-Z]{2}-[a-zA-Z]{2}\/");` and then `const path = location.href.split(regExp)[1];`

File: src/universal/common/components/AppHeader/AppHeader.tsx (hunk 6 above)
Comment (liviuignat) [inline src/universal/common/components/AppHeader/AppHeader.tsx]: Yes, damn, good catch, it should be `.co`. Btw `everreal.de` is pointed to redirect to `everreal.co`, but `https://www.everreal.de` fails to load as far as I see now.

File: src/translations/locale/locale.en-US.json (hunk 2 above)
Comment (RaulTsc) [inline src/translations/locale/locale.en-US.json]: Could be in configuration, but I think it's better to determine it programatically. I assume the auth plugin will always be on the same domain at the url `auth.baseUrl`, so then you can easily get `baseUrl`

Comment (RaulTsc) [inline src/translations/locale/locale.en-US.json]: Maybe something like `const baseUrl = location.href.replace("auth", "www");`

File: src/universal/common/components/AppFooter/AppFooter.tsx (hunk 4 above)
Comment (RaulTsc) [inline src/universal/common/components/AppFooter/AppFooter.tsx]: This will fail on something like `https://www.everreal.co/en-US/test/te-st/` though. Maybe split by `const regExp = new RegExp("{baseUrl}\/[a-zA-Z]{2}-[a-zA-Z]{2}\/");`

Comment (RaulTsc) [inline src/universal/common/components/AppFooter/AppFooter.tsx]: But we can't use location as it's rendered server-side... right? :/

Comment (liviuignat) [inline src/universal/common/components/AppFooter/AppFooter.tsx]: I am thinking that maybe something like `location.href.replace(currentLanguage, 'de-DE')` could do the trick also, but the current language should be read from `route.params.language`, not from `this.context.intl.locale`, because an url like `/whatever/login/sub-route` will have `route.params.language === 'whatever'` and `this.context.intl.locale === defaultLanguage === 'en-US'`.

Comment (liviuignat) [inline src/universal/common/components/AppFooter/AppFooter.tsx]: Ah right.

Comment (RaulTsc) [inline src/universal/common/components/AppFooter/AppFooter.tsx]: Lol, I quickly deleted my comment thinking you know something I don't :)

Comment (liviuignat) [inline src/universal/common/components/AppFooter/AppFooter.tsx]: I know that `redux-router` provides the location in the store or in props, it should be able to read `request.url` or `location.href`.

Comment (liviuignat) [inline src/universal/common/components/AppFooter/AppFooter.tsx]: I think this should be solved in another PR ... will add a ticket and create another PR.

Comment (RaulTsc) [inline src/universal/common/components/AppFooter/AppFooter.tsx]: Cool. If we do this we should do it for all links (AppLogo & About Us & others)

**Comment by liviuignat** [comment]
Ok, fixed the typos and added a new ticket for all the dynamic url changes we talked about. Will merge.
